### PR TITLE
MM-43276/MM-43247/MM-43248 Adding fields for DAU/WAU/MAU calculations and Stripe Subscriptions

### PR DESCRIPTION
**Pull request by sri-byte**
**Impact**: Adding fields for DAU/WAU/MAU calculations, Stripe Subscriptions, and unhiding IP address and Location fields.

a) DAU/WAU/MAU calculations:
 - https://mattermost.atlassian.net/browse/MM-43247
 - https://mattermost.atlassian.net/browse/MM-43248

b) Adding Plan field from Stripe Subscriptions, the same information can be derived from product_id, but I'd like to use the information from the field directly than deriving it.

~~c) Unhiding, IP Address and Location fields from server_daily_details for internal purposes. It isn't being used anywhere at the moment, but could be useful for future analyses.~~

d) Adding password_requirements field to user_events_telemetry for https://mattermost.atlassian.net/browse/MM-43276

**Testing** - Changes were tested and the visualization was added to the Growth Scorecard. The changes will be visible to the broader audience once this PR is merged.

### CODE REVIEW

**Comment by jasonblais** [comment]
@sri-byte Can you help link to the Jira tickets these changes are for, and add any testing notes?

**Comment by jasonblais** [comment]
@sri-byte Friendly reminder to help link to the Jira tickets these changes are for, and to add any testing notes?

File: views/mattermost/server_daily_details.view.lkml [hunk 1]
```diff
@@ -374,7 +374,7 @@ view: server_daily_details {
     description: "The server's IP Address."
     type: string
     sql: ${TABLE}.ip_address ;;
-    hidden: yes
+    hidden: no
```
Comment (jasonblais) [inline views/mattermost/server_daily_details.view.lkml:377]: Would we want to keep these hidden?

**Review by jasonblais — CHANGES_REQUESTED** [review]

File: views/mattermost/server_daily_details.view.lkml (hunk 1 above)
Comment (sri-byte) [reply views/mattermost/server_daily_details.view.lkml:377]: It doesn't matter if its hidden or not, but I want to unhide them.

**Comment by sri-byte** [comment]
> @sri-byte Friendly reminder to help link to the Jira tickets these changes are for, and to add any testing notes?

Updated the description.

File: views/stripe/subscriptions.view.lkml [hunk 2]
```diff
@@ -120,6 +120,11 @@ view: subscriptions {
     sql: ${TABLE}."QUANTITY" ;;
   }
 
+  dimension: plan {
+    type: string
+    sql: COALESCE(${TABLE}."PLAN":product, ${TABLE}."METADATA":current_product_id) ;;
```
Comment (jasonblais) [inline views/stripe/subscriptions.view.lkml:125]: What Jira ticket is this associated with?

**Comment by jasonblais** [comment]
@sri-byte if you can help link to the Jira tickets associated with hiding the IP_address and Location fields + the Stripe subscription changes - or if one doesn't exist, include a brief summary in the GitHub PR description. That way others have the right context on the changes, if someone looks back at this PR.

Finally, my one worry with exposing IP Address is that I'm not sure if that is personally identifiable information. Potentially something we'd need to check with security/privacy team (Daniel Schalla).

File: views/stripe/subscriptions.view.lkml (hunk 2 above)
Comment (sri-byte) [reply views/stripe/subscriptions.view.lkml:125]: this isn't associated to a Jira ticket, the information this field is bringing is already available in product_id, but that field is missing some data. I'd like to use this field over product_id.

Comment (jasonblais) [reply views/stripe/subscriptions.view.lkml:125]: Thanks @sri-byte, that works :+1:  Let's add it to the PR description so others have the context on the changes (if they happen to look at the PR at a later date).

**Review by jasonblais — APPROVED** [review]

**Pull request merged**